test/rhTable.js

```diff
@@ -0,0 +1,139 @@
+    } );
+
+    it( "should have defaults with data", function ()
+    {
+        utils.set( "list", [ { name: "one", id: 1 }, { name: "two", id: 2 } ] );
+
+        expect( $$( ".rh-table thead th" ).count() ).toBe( 3 ); // 3 columns
+        expect( $$( ".rh-table thead tr" ).count() ).toBe( 1 ); // 1 header row
+        expect( $$( ".rh-table tbody tr" ).count() ).toBe( 2 ); // 2 body rows
+        expect( $$( ".rh-table tbody tr:first-child .glyphicon-edit" ).count() ).toBe( 1 ); // edit button
+        expect( $$( ".table-controls" ).count() ).toBe( 1 ); // add button visible
+    } );
+
+    it( "should have non-view columns", function ()
+    {
+        utils.set( "definition", { meta: {}, id: { canViewList: false }, name: {} } );
+
+        expect( $$( ".rh-table thead th" ).count() ).toBe( 2 ); // 2 columns
+    } );
+
+    it( "should be able to open Add an item", function ()
+    {
+        utils
+            .expectNgDialogIsHidden()
+            .addForm()
+            .expectNgDialogIsVisible();
+    } );
+
+    it( "should cancel adding an item", function ()
+    {
+        $( ".ngdialog .initial" ).element( by.buttonText( "Cancel" ) ).click();
+        browser.sleep( 1000 );
+        utils.expectNgDialogIsHidden();
+    } );
+
+    it( "should save a new item", function ()
+    {
+        utils.addForm();
+        $( ".ngdialog .initial" ).element( by.buttonText( "Save" ) ).click();
+        browser.sleep( 1000 );
+        utils.expectNgDialogIsHidden();
+        expect( $$( ".rh-table tbody tr" ).count() ).toBe( 3 ); // 3 body rows
+    } );
+
+    it( "should edit an item", function ()
+    {
+        expect( $$( ".rh-table tbody tr.updated" ).count() ).toBe( 1 ); // 1 updated rows
+        expect( $( ".rh-table tbody tr:first-child td:first-child" ).getText() ).toBe( "one" );
+
+        $$( ".rh-table tbody tr .glyphicon-edit" ).get( 0 ).click();
+        $( ".ngdialog" ).element( by.model( "model.name" ) ).sendKeys( "-edited" );
+        $( ".ngdialog" ).element( by.buttonText( "Save" ) ).click();
+        browser.sleep( 1000 );
+
+        expect( $( ".rh-table tbody tr:first-child td:first-child" ).getText() ).toBe( "one-edited" );
+        utils.expectNgDialogIsHidden();
+        expect( $$( ".rh-table tbody tr.updated" ).count() ).toBe( 2 ); // 2 updated rows
+    } );
+
+    it( "should delete an item", function ()
+    {
+        expect( $$( ".rh-table tbody tr.deleted" ).count() ).toBe( 0 ); // no deleted rows
+        expect( $( ".ngdialog .delete-confirm" ).isPresent() ).toBe( false );
+
+        $$( ".rh-table tbody tr .glyphicon-edit" ).get( 2 ).click();
+        $( ".ngdialog .initial" ).element( by.buttonText( "Delete" ) ).click();
+        $( ".ngdialog .delete-confirm" ).element( by.buttonText( "Yes, Delete" ) ).click();
+        browser.sleep( 1000 );
+
+        utils.expectNgDialogIsHidden();
+        expect( $$( ".rh-table tbody tr.deleted" ).count() ).toBe( 1 ); // 1 deleted row
```

Comment:
I found it to be somewhat disorienting to just have it be gone because I'd forget which row I was on